querydsl-r2dbc/src/main/java/com/querydsl/r2dbc/H2Templates.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2015, The Querydsl Team (http://www.querydsl.com/team)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.querydsl.r2dbc;
+
+import com.querydsl.core.types.Ops;
+
+import java.sql.Types;
+
+/**
+ * {@code H2Templates} is an SQL dialect for H2
+ *
+ * @author tiwe
+ */
+public class H2Templates extends SQLTemplates {
```

Comment:
H2 templates is on the classpath (you're depending on `querydsl-sql` ). Why is there a need to also ship it here?

I would maybe like to see the `H2RdbcTemplates` extend the actual `H2Templates`. At least then we do not have to worry about the two template variants diverging too much leading to bugs.

But is it really necessary? I am fine with extending the `SQLTemplates`  API in `querydsl-sql`  with a couple of methods that you might need for proper R2dbc support.

Reply:
the reason behind this is that the underlying super `SQLTemplates` differs because of a single line `com.querydsl.r2dbc.SQLTemplates#serializeInsert:919` (`R2DBCInsertBatch` vs `SQLInsertBatch`), yet again didn't know how invasive I'm allowed to be